src/riak_kv_index_hashtree.erl

```diff
@@ -296,11 +302,12 @@ handle_call({delete, Items}, _From, State) ->
 handle_call(get_trees, _From, #state{trees=Trees}=State) ->
     {reply, Trees, State};
 
-handle_call({update_tree, Id}, From, State) ->
+handle_call({update_tree, Id, Callback}, From, State) ->
     lager:debug("Updating tree: (vnode)=~p (preflist)=~p", [State#state.index, Id]),
     apply_tree(Id,
                fun(Tree) ->
                        {SnapTree, Tree2} = hashtree:update_snapshot(Tree),
+                       catch Callback(),
```

Comment:
Wonder if perhaps we shouldn't move this inside the existing spawn_link() call below, so that the callback can't unnecessarily block the hashtree either?